### PR TITLE
Clone or New Asset: provide an initial schema name in a modal

### PR DESCRIPTION
**Pull request by jobelenus**
This is in anticipation of making the schema name immutable after creation.

### CODE REVIEW

File: lib/dal/src/schema/variant/authoring.rs [hunk 1]
```diff
@@ -350,10 +348,10 @@ impl VariantAuthoringClient {
             )
             .await?;
 
-        let schema_spec = metadata.to_spec(new_variant_spec)?;
+        let schema_spec = metadata.to_schema_spec(new_variant_spec)?;
         //TODO @stack72 - figure out how we get the current user in this!
         let pkg_spec = PkgSpec::builder()
-            .name(&metadata.name)
+            .name(&metadata.schema_name)
```
Comment (jobelenus) [inline lib/dal/src/schema/variant/authoring.rs:354]: this is the value from line 332 `name`, right? just trying to follow 😅

File: lib/sdf-server/src/server/service/variant/create_variant.rs [hunk 2]
```diff
@@ -38,9 +37,15 @@ pub async fn create_variant(
 ) -> SchemaVariantResult<impl IntoResponse> {
     let mut ctx = builder.build(request_ctx.build(request.visibility)).await?;
 
+    if Schema::is_name_taken(&ctx, &request.name).await? {
```
Comment (jobelenus) [inline lib/sdf-server/src/server/service/variant/create_variant.rs:40]: ✌️

File: lib/dal/src/schema/variant/authoring.rs (hunk 1 above)
Comment (vbustamante) [reply lib/dal/src/schema/variant/authoring.rs:354]: Yup, it gets packaged into the metadata object

**Comment by vbustamante** [comment]
bors merge

**Pull request merged**